### PR TITLE
Update CI/CD setup

### PR DESCRIPTION
**Pull request by dnlkoch**
This updates all packages, removes some outdated ones and applies some changes to the CI setup for the workshop by:

- applying `commitlint` and `semantic-release`
- adding a test pipeline
- replacing `gh-pages` with `JamesIves/github-pages-deploy-action`

Please review @terrestris/devs.

### CODE REVIEW

**Comment by hblitza** [comment]
LGTM

**Review by ahennr — APPROVED** [review]

**Review by hblitza — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -1,34 +1,35 @@
...
   "bugs": {
     "url": "https://github.com/terrestris/geoserver-introduction-ws/issues"
   },
-  "homepage": "https://github.com/terrestris/geoserver-introduction-ws#readme",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/terrestris/geoserver-introduction-ws.git"
+  },
+  "license": "MIT",
+  "author": "terrestris GmbH & Co.KG",
```
Comment (LukasLohoff) [inline package.json]:
```suggestion
  "author": "terrestris GmbH & Co. KG",
```

**Review by LukasLohoff — APPROVED** [review]

**Review by hwbllmnn — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (dnlkoch) [reply package.json]: Nice catch!

**Comment by dnlkoch** [comment]
Thanks for the reviews!

**Pull request merged**